mPowerSDK/Dashboard/APHDashboardGraphTableViewCell.m

```diff
@@ -24,14 +30,162 @@ + (CGFloat)medicationLegendContainerHeight
     return kMedicationLegendContainerHeight;
 }
 
++ (CGFloat)sparkLineGraphContainerHeight
+{
+    return kSparkLineGraphContainerHeight;
+}
+
++ (CGFloat)correlationSelectorHeight
+{
+    return kCorrelationSelectorHeight;
+}
+
+- (void)awakeFromNib
+{
+    [super awakeFromNib];
+    
+    [self giveImageViewDownCarrotImage:self.button1DownCarrot];
+    [self giveImageViewDownCarrotImage:self.button2DownCarrot];
+    
+    [self updateSegmentColors];
+}
+
+#pragma mark - Helper Methods
+
+- (void)giveImageViewDownCarrotImage:(UIImageView *)button
+{
+    [button setImage:[UIImage imageNamed:@"down_carrot"]];
+    [button setTintColor:[UIColor lightGrayColor]];
+}
+
+// From: http://stackoverflow.com/questions/1196679/customizing-the-colors-of-a-uisegmentedcontrol
+- (void) updateSegmentColors
+{
+    NSUInteger numSegments = [self.correlationSegmentControl.subviews count];
+    
+    // Reset segment's color
+    for( int i = 0; i < numSegments; i++ ) {
+        [[self.correlationSegmentControl.subviews objectAtIndex:i] setTintColor:nil];
+        [[self.correlationSegmentControl.subviews objectAtIndex:i] setTintColor:[UIColor appSecondaryColor2]];
+        
+        UIView *segmentView = [self.correlationSegmentControl.subviews objectAtIndex:i];
+        for (UIImageView *imageView in segmentView.subviews) {
+            [imageView setTintColor:[UIColor appTertiaryGrayColor]];
+        }
+    }
+    
+    // Sort Segments from left to right
+    NSArray *sortedViews = [self.correlationSegmentControl.subviews sortedArrayUsingFunction:compareViewsByOrigin context:NULL];
+    
+    // Set selected segment color
+    NSInteger selectedIdx = self.correlationSegmentControl.selectedSegmentIndex;
+    UIColor *darkBlueColor = [UIColor colorWithRed:0.24 green:0.37 blue:0.51 alpha:1];
+    [[sortedViews objectAtIndex:selectedIdx] setTintColor:darkBlueColor];
+    
+    // Remove all original segments from the control
+    for (id view in self.correlationSegmentControl.subviews) {
+        [view removeFromSuperview];
+    }
+    
+    // Append sorted and colored segments to the control
+    for (id view in sortedViews) {
+        [self.correlationSegmentControl addSubview:view];
+    }
+}
+
+
+NSInteger static compareViewsByOrigin(id sp1, id sp2, void *context)
+{
+    float v1 = ((UIView *)sp1).frame.origin.x;
+    float v2 = ((UIView *)sp2).frame.origin.x;
+    if (v1 < v2)
```

Comment:
Nit: the absence of curly braces can result in failures when the code needs to change.